src/presentation/device.rs

```diff
@@ -542,22 +548,15 @@ pub trait DeviceSession {
                     continue;
                 }
             };
-            let prepared_cose_sign1 = match CoseSign1::builder()
-                .detached()
-                .payload(device_auth_bytes)
-                .signature_algorithm(signature_algorithm)
-                .prepare()
-            {
-                Ok(prepared) => prepared,
-                Err(_e) => {
-                    let error: DocumentError = [(doc_type, DocumentErrorCode::DataNotReturned)]
-                        .into_iter()
-                        .collect();
-                    document_errors.push(error);
-                    continue;
-                }
+            let protected = Header {
+                alg: Some(RegisteredLabelWithPrivate::Assigned(signature_algorithm)),
+                ..Header::default()
             };
-
+            // todo: support for CoseMac0
+            let builder = CoseSign1Builder::new()
+                .protected(protected)
+                .payload(device_auth_bytes);
+            let prepared_cose_sign1 = CoseSign1::new(builder.build());
```

Comment:
The payload is supposed to be detached here, so will need to be removed on line 409. Consider making a `PreparedCoseSign1` struct to make it harder to make this mistake.